### ISSUE TITLE
Metadata must be an object

### ISSUE THREAD

**Issue by ericzbeard** [aws-cloudformation/cfn-lint#2758]
### CloudFormation Lint Version

cfn-lint 0.72.10

### What operating system are you using?

Mac

### Describe the bug

Cfn-lint should error on Metadata that is not an object.

### Expected behavior

Fail with an error when Metadata is not an object.

### Reproduction template

```yaml
Resources:
    Something:
        Metadata: Some text...
```

Should be

```yaml
Resources:
    Something:
        Metadata:
            Comment: Some text...
```

Error from CloudFormation:

```
error creating changeset: Template format error: Any Metadata member must be an object.
```

**Comment by kddejong** [comment]
This will be fixed in v1.  Going to close this one.
